pkg/api/norman/customization/aks/listers.go

```diff
@@ -120,6 +123,92 @@ func (s sortableVersion) Less(a, b int) bool {
 	return version.Compare(s[a], s[b], "<")
 }
 
+type KubernetesUpgradeVersion struct {
+	Version string `json:"version"`
+	Enabled bool   `json:"enabled"`
+}
+
+type KubernetesUpgradeVersions []*KubernetesUpgradeVersion
+
+func (s KubernetesUpgradeVersions) Len() int {
+	return len(s)
+}
+
+func (s KubernetesUpgradeVersions) Swap(a, b int) {
+	s[a], s[b] = s[b], s[a]
+}
+
+func (s KubernetesUpgradeVersions) Less(a, b int) bool {
+	return version.Compare(s[a].Version, s[b].Version, "<")
+}
+
+// listKubernetesUpgradeVersions lists all kubernetes versions listed by AKS Container Service and marks which ones the
+// given cluster can be upgraded to.  A version's `Enabled` flag is true if the cluster can be upgraded to the version
+// in its current state.
+func listKubernetesUpgradeVersions(ctx context.Context, clusterLister v3.ClusterLister, cap *Capabilities) ([]byte, int, error) {
+	// load the target cluster, if the cluster is not found we cannot proceed
+	cluster, err := clusterLister.Get("", cap.ClusterID)
+	if err != nil {
+		return nil, http.StatusBadRequest, fmt.Errorf("failed to load cluster: %v", err)
+	}
+
+	// get the client for aks container service
+	clientContainer, err := NewContainerServiceClient(cap)
+	if err != nil {
+		return nil, http.StatusInternalServerError, err
+	}
+
+	// request a list of orchestrators
+	orchestrators, err := clientContainer.ListOrchestrators(ctx, cap.ResourceLocation, "managedClusters")
+	if err != nil {
+		return nil, http.StatusBadRequest, fmt.Errorf("failed to get orchestrators: %v", err)
+	}
+
+	// ensure the orchestrators are returned
+	if orchestrators.Orchestrators == nil {
+		return nil, http.StatusBadRequest, fmt.Errorf("no version profiles returned: %v", err)
+	}
+
+	versions := make(KubernetesUpgradeVersions, 0)
+	upgradeVersions := make(map[string]bool, 0)
+	for _, profile := range *orchestrators.Orchestrators {
+		// check for nil pointers to avoid a panic
+		if profile.OrchestratorType == nil || profile.OrchestratorVersion == nil {
+			logrus.Debug("unexpected nil orchestrator type or version")
+			continue
+		}
+
+		// filter down to only kubernetes types
+		if containerservice.OrchestratorTypes(*profile.OrchestratorType) == containerservice.Kubernetes {
+			// store this kubernetes version
+			versions = append(versions, &KubernetesUpgradeVersion{Version: *profile.OrchestratorVersion})
+
+			// generate the upgrade map when the current version is found
+			if *profile.OrchestratorVersion == *cluster.Spec.AKSConfig.KubernetesVersion {
+				upgradeVersions = buildUpgradeableVersionsMap(profile)
+			}
+		}
+	}
+
+	// disable any version not listed in the upgrade versions
```

Comment:
This comment needs to be updated to match the change in implementation (setting Enabled to `true` vs. `false` as before)